src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,302 @@
+        printOutput(output);
+    }
+
+    private static void displayUnknownCommandResponse() {
+        String unknownCommandResponse = " ☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n";
+        printOutput(unknownCommandResponse);
+    }
+
+    private static void executeCommand(String command, String arguments) throws DukeException {
+        switch (command) {
+        case "list":
+            listTasks();
+            break;
+        case "done":
+            markDone(arguments);
+            break;
+        case "todo":
+            addTodoTask(arguments);
+            break;
+        case "deadline":
+            addDeadlineTask(arguments);
+            break;
+        case "event":
+            addEventTask(arguments);
+            break;
+        case "delete":
+            deleteTask(arguments);
+            break;
+        case "bye":
+            saveData();
+            displayByeMessage();
+            System.exit(0);
+        default:
+            throw new DukeException();
+        }
+    }
+
+    private static void writeToFile(String data) throws IOException {
+        FileWriter fw = new FileWriter("data/duke.txt");
+        fw.write(data);
+        fw.close();
+    }
+
+    private static void saveData() {
+        String data = "";
+        for (Task task : tasks) {
+            data += task.toFileFormat() + System.lineSeparator();
+        }
+
+        File myDirectory = new File("data");
+        File myFile = new File("data/duke.txt");
+
+        if (!myDirectory.exists()) {
+            myDirectory.mkdir();
+        }
+
+        try {
+            myFile.createNewFile();
+            writeToFile(data);
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void readFromFile(File myFile) throws FileNotFoundException {
+        Scanner sc = new Scanner(myFile);
+        while (sc.hasNext()) {
+            String taskDetails = sc.nextLine();
+
+            String[] splitTaskDetails = taskDetails.split("\\|");
+            String taskType = splitTaskDetails[0];
+            Boolean taskStatus = false;
+            if (splitTaskDetails[1].equals("true")) {
+                taskStatus = true;
+            }
+            String description = splitTaskDetails[2];
+            String date = "";
+            if (splitTaskDetails.length > 3) {
+                date = splitTaskDetails[3];
+            }
+
+            switch (taskType) {
+            case "T":
+                Todo newTodo = new Todo(description);
+                tasks.add(newTodo);
+                if (taskStatus) {
+                    newTodo.markAsDone();
+                }
+                break;
+            case "D":
+                Deadline newDeadline = new Deadline(description, date);
+                tasks.add(newDeadline);
+                if (taskStatus) {
+                    newDeadline.markAsDone();
+                }
```

Comment:
You may want to put this at the end of the switch so you don't end up repeating the same statement 3 times.